model/content_entity.go

```diff
@@ -749,3 +751,44 @@ func (w *ContentEntity) UnmarshalJSON(data []byte) error {
 	err = json.Unmarshal(data, &w.Property)
 	return err
 }
+
+//GenerateID adds a generated id to the payload based on the schema
+func GenerateID(payload []byte, entityFactory EntityFactory) ([]byte, error) {
+	schema := entityFactory.Schema()
+	properties := entityFactory.Schema().ExtensionProps.Extensions["x-identifier"]
```

Comment:
though passing in the schema is not needed, you set a variable "schema" then you still use the schema on the entity factory?